src/main/java/ServiceImpl/AddressDAOImpl.java

```diff
@@ -0,0 +1,44 @@
+package ServiceImpl;
+
+import Dao.AddressDao;
+import Entities.Address;
+import Services.DAOService;
+
+import java.util.List;
+
+public class AddressDAOImpl implements DAOService<Address> {
+
+    AddressDao addressDao = null;
+

```

Comment:
Good work! I have only one thing about Service layer classes. I think Service layer classes should work to make a connection/interaction between entity classes. The rest looks great!

Reply:
Thankyou for approving my code, can you please explain more on service layer making connection to entity classes? I think service layer is used to prevent coupling (encourage decoupling) between application/UI/presentation from database layer.